cirq/work/collector.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+from typing import Optional, Any, Iterable, Union, List
+
+import numpy as np
+
+from cirq import circuits, study, value
+from cirq.work import sampler, work_pool
+
+
+@value.value_equality(unhashable=True)
+class CircuitSampleJob:
+    """Describes a sampling task."""
+
+    # pylint: disable=redefined-builtin
+    def __init__(self,
+                 circuit: circuits.Circuit,
+                 *,
+                 repetitions: int,
+                 id: Any = None):
```

Comment:
Maybe call this `tag` or `label` to avoid redefining the `id` builtin?

Reply:
Done.